rioxarray/_io.py

```diff
@@ -0,0 +1,392 @@
+"""
+
+Credits:
+
+This file was adopted from: https://github.com/pydata/xarray # noqa
+Source file: https://github.com/pydata/xarray/blob/1d7bcbdc75b6d556c04e2c7d7a042e4379e15303/xarray/backends/rasterio_.py # noqa
+"""
+
+import os
+import warnings
+from collections import OrderedDict
+from distutils.version import LooseVersion
+
+import numpy as np
+from xarray import DataArray
+from xarray.core import indexing
+from xarray.core.utils import is_scalar
+from xarray.backends.common import BackendArray
+from xarray.backends.file_manager import CachingFileManager
+from xarray.backends.locks import SerializableLock
+
+from rioxarray.rioxarray import affine_to_coords
+
+
+# TODO: should this be GDAL_LOCK instead?
+RASTERIO_LOCK = SerializableLock()
+
+_ERROR_MSG = (
+    "The kind of indexing operation you are trying to do is not "
+    "valid on rasterio files. Try to load your data with ds.load()"
```

Comment:
Extra space needed at the end here.

Reply:
Probably just remove. It appears unused.